main/src/main/java/org/mobilitydata/gtfsvalidator/validator/LevelPresenceValidator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import javax.inject.Inject;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.SeverityLevel;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsLevelTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsPathway;
+import org.mobilitydata.gtfsvalidator.table.GtfsPathwayMode;
+import org.mobilitydata.gtfsvalidator.table.GtfsPathwayTableContainer;
+
+/**
+ * Checks that levels.txt is provided if at least a row from pathways.txt has {@code
+ * pathway.pathway_mode=5}.
+ *
+ * <p>Generated notice: {@link MissingLevelNotice}.
+ */
+@GtfsValidator
+public class LevelPresenceValidator extends FileValidator {
+
+  private final GtfsLevelTableContainer levels;
+  private final GtfsPathwayTableContainer pathways;
+
+  @Inject
+  LevelPresenceValidator(GtfsLevelTableContainer levels, GtfsPathwayTableContainer pathways) {
+    this.levels = levels;
+    this.pathways = pathways;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    if (levels.entityCount() != 0) {
+      return;
+    }
+    ListMultimap<GtfsPathwayMode, GtfsPathway> byPathwayModeMap = ArrayListMultimap.create();
+    for (GtfsPathway pathway : pathways.getEntities()) {
+      byPathwayModeMap.put(pathway.pathwayMode(), pathway);
+    }
+    if (byPathwayModeMap.get(GtfsPathwayMode.ELEVATOR) == null) {
+      return;
+    }
+    GtfsPathway pathway = byPathwayModeMap.get(GtfsPathwayMode.ELEVATOR).get(0);
```

Comment:
Are selecting only the first faulty pathway because we add only 1 notice when this problem occurs? If yes and this is what we want, it looks good to me. 
My concern with this: what happens if the problem comes from the pathways, i.e. there is no elevator at all, but the data producer have mixed up the pathways modes? In such case, generating a single notice for the first pathway won't help the data producer fixing the issue (changing the pathway mode of every faulty pathway record).

Reply:
> Are selecting only the first faulty pathway because we add only 1 notice when this problem occurs? If yes and this is what we want, it looks good to me.

Exactly.

> My concern with this: what happens if the problem comes from the pathways, i.e. there is no elevator at all, but the data producer have mixed up the pathways modes? In such case, generating a single notice for the first pathway won't help the data producer fixing the issue (changing the pathway mode of every faulty pathway record).

If this happens, a notice will be generated every time the data producer fixes a row from `pathways.txt` until all rows from said file have `pathway_mode != 5`.

Reply:
One thing I could do to improve this implementation is passing the list of pathway_id that have `pathway_mode=5`. So that the data producer/consumer has all the information without potentially generating too many notices that could lead to an OOM error.

Reply:
As discussed, I clarified the documentation to make it obvious that we assume that `pathways.txt` is correct.